common/djangoapps/entitlements/tests/test_models.py

```diff
@@ -0,0 +1,92 @@
+"""Test Entitlements models"""
+
+import unittest
+from datetime import datetime, timedelta
+
+import pytz
+from django.conf import settings
+from django.test import TestCase
+
+from openedx.core.djangoapps.content.course_overviews.tests.factories import CourseOverviewFactory
+from student.tests.factories import CourseEnrollmentFactory
+
+# Entitlements is not in CMS' INSTALLED_APPS so these imports will error during test collection
+if settings.ROOT_URLCONF == 'lms.urls':
+    from entitlements.tests.factories import CourseEntitlementFactory
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class TestModels(TestCase):
+    """Test entitlement with policy model functions."""
+
+    def setUp(self):
+        super(TestModels, self).setUp()
+        self.course = CourseOverviewFactory.create(
+            start=datetime.utcnow()
+        )
+        self.enrollment = CourseEnrollmentFactory.create(course_id=self.course.id)
+
+    def test_is_entitlement_redeemable(self):
+        """
+        Test that the entitlement is not expired when created now, and is expired when created two years
+        ago with a policy that sets the expiration period to 450 days
+        """
+
+        entitlement = CourseEntitlementFactory.create()
+
+        assert entitlement.is_entitlement_redeemable() is True
+
+        # Create a date 2 years in the past (greater than the policy expire period of 450 days)
+        past_datetime = datetime.utcnow().replace(tzinfo=pytz.UTC) - timedelta(days=365 * 2)
+        entitlement.created = past_datetime
+        entitlement.save()
+
+        assert entitlement.is_entitlement_redeemable() is False
+
+    def test_is_entitlement_refundable(self):
+        """
+        Test that the entitlement is refundable when created now, and is not refundable when created two years
```

Comment:
Nit, we should check that it is not refundable when created n days ago, where n is < expiration_period_days but > refund_period_days. Otherwise, we cannot tell from the test whether it's passing because the entitlement expired, or because the policy properly applied the refund period.

Reply:
Nit: this should say 'when created 70 days ago'